experiments/state/src/main/scala/org/apache/gearpump/streaming/state/system/impl/UnboundedState.scala

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.gearpump.streaming.state.system.impl
+
+import com.twitter.algebird.Monoid
+import org.apache.gearpump.TimeStamp
+import org.apache.gearpump.streaming.state.system.api.MonoidState
+import org.apache.gearpump.util.LogUtil
+import org.slf4j.Logger
+
+object UnboundedState {
+  val LOG: Logger = LogUtil.getLogger(classOf[UnboundedState[_]])
+
+  def apply[T](implicit monoid: Monoid[T]): UnboundedState[T] = {
+    new UnboundedState(monoid)
+  }
+}
+
+/**
+ * this is a MonoidState with no boundary
+ */
+class UnboundedState[T](monoid: Monoid[T]) extends MonoidState[T](monoid) {
```

Comment:
Addition.

I don't like the word boundary, as I don't know what boundary means here, time boundary? I know you want to differentiate from WindowState, maybe NonWindowState better?